inspire_disambiguation/api.py

```diff
@@ -164,6 +184,10 @@ def cluster(ethnicity_model_path, distance_model_path, n_jobs, signature_block=N
             B3_f_score=cluster.supervised_scoring(clusterer.y, cluster.labels_)
             if hasattr(cluster, "supervised_scoring")
             else None,
+            # should I check for the supervised_scoring here too?
+            B3_f_score_validation=cluster.score(validation_signatures_uuids, cluster.labels_)
```

Comment:
I think you're missing a step here, which is to remove the labels from those signatures, otherwise you're definitely using them in the computation of the optimal cutting distance.